src/main/resources/avro/referencemethods.avdl

```diff
@@ -325,4 +386,99 @@ GetSequenceBasesResponse getSequenceBases(
     /** Additional request parameters to restrict the query. */
     GetSequenceBasesRequest request) throws GAException;
 
+/****************  /joins/search  *******************/
+/**
+This request maps to the body of `POST /joins/search` as JSON. Specifies a
+number of filters, all of which must be satisfird by each result returned.
+*/
+record SearchJoinsRequest {
+  /**
+  If present, return only `Join`s which belong to this reference set.
+  */
+  union { null, string } referenceSetId = null;
+
+  /**
+  If present, return only `Join`s which belong to this variant set.
+  */
+  union { null, string } variantSetId = null;
+
+  // TODO: Specify a proper sort of filter object instead of this. We want to be
+  // able to support a list of sequences, but also a single sequence and a
+  // range.
+
```

Comment:
@jeromekelleher the `SearchJoinsRequest` is intended to return a subset of joins present in a given (graph) reference or variant set, allowing the client to re-construct (a portion of) the graph from a list of joins and sequences. The `starts`, `lengths` and `strands` options allow the client to restrict the list of returned joins to a particular subgraph of interest (say, starting at a particular known location on a given reference sequence for a gene with several different alleles represented by a moderately complex subgraph, and limited to the length of that gene's "allele subgraph" along that reference sequence). Since their intent is to (cumulatively) restrict the scope of the search, I'm not sure how splitting these out into separate API calls would help.